closed/src/java.base/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java

```diff
@@ -309,14 +318,16 @@ int decrypt(byte[] cipher, int cipherOffset, int cipherLen,
     int encryptFinal(byte[] plain, int plainOffset, int plainLen,
                      byte[] cipher, int cipherOffset) {
 
-        int ret = -1;
+        int ret;
 
-        if(plain == cipher) {
-            ret = nativeCrypto.CBCFinalEncrypt(nativeContext, plain.clone(),
-                                               plainOffset, plainLen, cipher, cipherOffset);
-        } else {
-            ret = nativeCrypto.CBCFinalEncrypt(nativeContext, plain, plainOffset,
-                                                plainLen, cipher, cipherOffset);
+        synchronized (this) {
+            if(plain == cipher) {
+                ret = nativeCrypto.CBCFinalEncrypt(nativeContext, plain.clone(),
+                                                plainOffset, plainLen, cipher, cipherOffset);
+            } else {
+                ret = nativeCrypto.CBCFinalEncrypt(nativeContext, plain, plainOffset,
+                                                    plainLen, cipher, cipherOffset);
+            }
```

Comment:
* formatting: space after if
* excessive and inconsistent indentation of continued arguments